tests/test_cold_demand.py

```diff
@@ -44,20 +46,21 @@ def test_insufficient_capacity(self):
                 profile_reader=ProfileReaderFromFile,
                 input_timeseries_file="timeseries.csv",
             )
-        # Is SystemExit is raised
-        np.testing.assert_array_equal(cm.exception.code, 1)
-
-        # Check that the heat & cold demand had an error
-        np.testing.assert_equal(
-            logs_list[0].msg == "HeatingDemand_9b90: The installed capacity of 0.05MW should be"
-            " larger than the maximum of the heat demand profile 0.15MW",
-            True,
-        )
-        np.testing.assert_equal(
-            logs_list[2].msg == "CoolingDemand_15e8: The installed capacity of 0.05MW should be"
-            " larger than the maximum of the heat demand profile 0.15MW",
-            True,
-        )
+            # Check that the cold demand had an error
+            np.testing.assert_equal(cm.error_type, MesidoAssetIssueType.COLD_DEMAND_POWER)
+            np.testing.assert_equal(
+                cm.general_issue,
+                "Asset insufficient installed capacity: please increase the installed power or"
+                " reduce the demand profile peak value of the demand(s) listed.",
+            )
+            np.testing.assert_equal(
+                cm.message_per_asset_id["15e803b4-1224-4cac-979f-87747a656741"],
+                "Asset named CoolingDemand_15e8: The installed capacity of 0.05MW should be larger"
+                " than the maximum of the heat demand profile 0.15MW",
+            )
+            # This is needed, else the POTENTIAL_ERRORS from this test still exists in the
+            # following tests
+            yield
```

Comment:
I tried removing the `yield` and the test passed. I don't expect a `yield` is necessary here. Could you provide some more info on what happens on your end?

Reply:
Without "yield": If you run "pytest tests/test_cold_demand.py" then the test fails when I run it locally. The reason being in "test_cold_demand.py" the test "test_insufficient_capacity()" is run 1st and error are captured and stored in POTENTIAL_ERRORS. The next test then runs "test_cold_demand()" and during this test the POTENTIAL_ERRORS._gathered_potential_issues exists but it was created in "test_insufficient_capacity()". So POTENTIAL_ERRORS is carried over from "test_insufficient_capacity()" to "test_cold_demand()".

Reply:
I also see yield creates other issues.

Reply:
Here is how to isolate a PotentialError instance within a single test: https://github.com/Multi-Energy-Systems-Optimization/mesido/pull/191/files

Reply:
Yield has been removed. And my issue has been resolved. Thanks. The code has been updated